lib/remote/apilistener-configsync.cpp

```diff
@@ -454,19 +501,17 @@ void ApiListener::SendRuntimeConfigObjects(const JsonRpcConnection::Ptr& aclient
 	Log(LogInformation, "ApiListener")
 		<< "Syncing runtime objects to endpoint '" << endpoint->GetName() << "'.";
 
+	std::unordered_set<ConfigObject*> syncedObjects;
 	for (const Type::Ptr& type : Type::GetAllTypes()) {
-		auto *dtype = dynamic_cast<ConfigType *>(type.get());
-
-		if (!dtype)
-			continue;
-
-		for (const ConfigObject::Ptr& object : dtype->GetObjects()) {
-			/* don't sync objects for non-matching parent-child zones */
-			if (!azone->CanAccessObject(object))
-				continue;
-
-			/* send the config object to the connected client */
-			UpdateConfigObject(object, nullptr, aclient);
+		if (auto *ctype = dynamic_cast<ConfigType *>(type.get())) {
+			for (const auto& object : ctype->GetObjects()) {
+				// All objects must be synced sorted by their dependency graph.
+				// Otherwise, downtimes/comments etc. might get synced before their respective Checkables, which will
+				// result in comments and downtimes being ignored by the other endpoint since it does not yet know
+				// about their checkables. Given that the runtime config updates event does not trigger a reload on the
+				// remote endpoint, these objects won't be synced again until the next reload.
+				UpdateConfigObjectWithParents(object, aclient, syncedObjects);
+			}
```

Comment:
## Optional

Admittedly I've reviewed the style of some changes you made and overseen that they just unnecessarily increase the diff in the first place. 😅

```suggestion
		auto *dtype = dynamic_cast<ConfigType *>(type.get());

		if (!dtype)
			continue;

		for (const ConfigObject::Ptr& object : dtype->GetObjects()) {
			// All objects must be synced sorted by their dependency graph.
			// Otherwise, downtimes/comments etc. might get synced before their respective Checkables, which will
			// result in comments and downtimes being ignored by the other endpoint since it does not yet know
			// about their checkables. Given that the runtime config updates event does not trigger a reload on the
			// remote endpoint, these objects won't be synced again until the next reload.
			UpdateConfigObjectWithParents(object, aclient, syncedObjects);
```

Reply:
If you had changed some _semantics_, I'd vote with both hands for a _while on it(TM)_. But I'm afraid it's a no-op diff. 🤷‍♂️